### DISCUSSION TITLE
Question: Use of current date in generated JSON

### DISCUSSION THREAD

**Discussion by eike-42** [rfennell/AzurePipelines#1259] [Q&A] [answered]
Hi,

first thanks for this great extension. This is automizing an important part of the release process!

One thing, I'd like to add the current date to the generated json file. Here is the file:
{
    "ReleaseVersion": "$(CurrentVersion)",
    "ReleaseDate": "$(Date)",
    "WorkItems": 
        [
            {{#forEach queryWorkItems}}
            {
                "ID":"{{this.id}}",
                "Name": "{{lookup this.fields 'System.Title'}}",
                "WIT": "{{lookup this.fields 'System.WorkItemType'}}",
                "Tags": "{{lookup this.fields 'System.Tags'}}",
            },
            {{/forEach}}
        ]
},

The $(CurrentVersion) works, the $(Date) does not. Is there a predefined variable for that which I can use? And can the date_formatter be used with it?

Thanks and have a great day
Eike

**Comment by rfennell** [comment, marked as answer]
There is no built in variable I am aware of. Best option is to write a custom extension/ Something like [this sample](https://github.com/rfennell/AzurePipelines/blob/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203/date_formatter.js) just in the current date as opposed to an injected one

**Reply by eike-42** [reply]
Thanks, that idea solved my problem!

module.exports = {getCurrentDate() {
    date=new Date();
    var d = date.getDate();
    var m = date.getMonth() + 1; //Month from 0 to 11
    var y = date.getFullYear();
    return '' + (d <= 9 ? '0' + d : d) + '.' + (m<=9 ? '0' + m : m) + '.' + y;
}};
